Lib/test/support/__init__.py

```diff
@@ -945,6 +945,27 @@ def check_sizeof(test, o, size):
             % (type(o), result, size)
     test.assertEqual(result, size, msg)
 
+def subTests(arg_names, arg_values, /, *, _do_cleanups=False):
+    """Run multiple subtests with different parameters.
+    """
+    if isinstance(arg_names, str):
+        arg_names = arg_names.split(',')
+    def decorator(func):
+        if isinstance(func, type):
+            raise TypeError('subTests() can only decorate methods, not classes')
+        @functools.wraps(func)
+        def wrapper(self, /, *args, **kwargs):
+            for values in arg_values:
+                if len(arg_names) == 1:
```

Comment:
I'd change the logic to *if the original arg_names was a string without a comma*.

That is, I'd expect `subTests(['number'], ([1], [2])` to set `number` to `1` and `2`.